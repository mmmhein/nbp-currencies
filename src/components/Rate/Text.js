import styled from 'styled-components';

import BaseText from '../Text';

const Text = styled(BaseText)`
  width: 100%;
`;

export default Text;
