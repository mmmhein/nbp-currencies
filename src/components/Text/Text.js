import styled from 'styled-components';

import media from '../../styles/media';
import { fontMedium, fontWeightNormal, fontWeightBold, colorBlack, fontSmall } from '../../styles/designTokens';

const Text = styled.p`
  font-size: ${fontMedium};
  line-height: 1.4;
  font-weight: ${({ bold }) => (bold ? fontWeightBold : fontWeightNormal)};
  margin: 0;
  color: ${colorBlack};

  ${media.phone`
    font-size: ${fontSmall};
  `};
`;

export default Text;

/*import styled from 'styled-components';

import media from '../../styles/media';
import { fontMedium, fontWeightNormal, fontWeightBold, colorBlack, fontSmall } from '../../styles/designTokens';

const Text = styled.p`
  font-size: ${fontMedium};
  font-weight: ${({ bold }) => (bold ? fontWeightBold : fontWeightNormal)};
  line-height: 1.4;
  margin: 0;
  color: ${colorBlack};

  ${media.phone`
    font-size: ${fontSmall}
  `};
`;

export default Text;
*/
