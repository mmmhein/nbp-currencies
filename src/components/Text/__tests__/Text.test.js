import React from 'react';
import { create } from 'react-test-renderer';

import Text from '../Text';

describe('COMPONENT - Text', () => {
  it('renders correctly', () => {
    const component = create(<Text />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly bold', () => {
    const component = create(<Text />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
