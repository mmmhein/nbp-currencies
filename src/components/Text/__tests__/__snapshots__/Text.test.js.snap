// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Text renders correctly 1`] = `
.c0 {
  font-size: 1.8rem;
  line-height: 1.4;
  font-weight: 400;
  margin: 0;
  color: #222222;
}

@media (max-width:26.25em) {
  .c0 {
    font-size: 1.4rem;
  }
}

<p
  className="c0"
/>
`;

exports[`COMPONENT - Text renders correctly bold 1`] = `
.c0 {
  font-size: 1.8rem;
  line-height: 1.4;
  font-weight: 400;
  margin: 0;
  color: #222222;
}

@media (max-width:26.25em) {
  .c0 {
    font-size: 1.4rem;
  }
}

<p
  className="c0"
/>
`;
